File: app/src/main/java/tommy_kw/esoutline/DesingInfo.java
package tommy_kw.esoutline;

import android.app.Activity;
import android.content.Intent;

import tommy_kw.esoutline.activity.CoordinatorActivity;
import tommy_kw.esoutline.activity.FloatingActionActivity;
import tommy_kw.esoutline.activity.TextInputLayoutActivity;

/**
 * Created by tomita on 15/06/25.
 */
public enum DesingInfo {
    CoordinatorLayout("CoordinatorLayout", CoordinatorActivity.class),
    FloatingActionLayout("FloatingActionLayout", FloatingActionActivity.class),
    TextInputLayout("TextInputLayout", TextInputLayoutActivity.class);

    private String mLabel;
    private Class<?> mActivity;

    DesingInfo(String label, Class<?> activity) {
        mLabel = label;
        mActivity = activity;
    }

    private Intent createdIntent(Activity activity) {
        return new Intent(activity, mActivity);
    }

    public String getLabel() {
        return mLabel;
    }

    public void startActivity(Activity activity) {
        activity.startActivity(createdIntent(activity));
    }
}

File: app/src/main/java/tommy_kw/esoutline/DesingInfo.kt
package tommy_kw.esoutline

import android.app.Activity
import android.content.Intent

import tommy_kw.esoutline.activity.CoordinatorActivity
import tommy_kw.esoutline.activity.FloatingActionActivity
import tommy_kw.esoutline.activity.TextInputLayoutActivity

/**
 * Created by tomita on 15/06/25.
 */
enum class DesingInfo private constructor(val label: String,
                                          val mActivity: Class<*>) {
    CoordinatorLayout("CoordinatorLayout", CoordinatorActivity::class.java),
    FloatingActionLayout("FloatingActionLayout", FloatingActionActivity::class.java),
    TextInputLayout("TextInputLayout", TextInputLayoutActivity::class.java);

    private fun createdIntent(activity: Activity): Intent {
        return Intent(activity, mActivity)
    }

    fun startActivity(activity: Activity) {
        activity.startActivity(createdIntent(activity))
    }
}
